packages/core/entity/FileDetail/index.ts

```diff
@@ -160,24 +161,36 @@ export default class FileDetail {
         return this.fileDetail.annotations.find((annotation) => annotation.name === annotationName);
     }
 
-    public getPathToThumbnail(): string | undefined {
+    public async getPathToThumbnail(): Promise<string | undefined> {
+        // If the thumbnail is a relative path on the allen drive then preprend it to
+        // the AICS FMS NGINX server path
+        if (this.thumbnail?.startsWith("/allen")) {
+            return `${AICS_FMS_FILES_NGINX_SERVER}${this.thumbnail}`;
+        }
+
         // If no thumbnail present try to render the file itself as the thumbnail
         if (!this.thumbnail) {
             const isFileRenderableImage = RENDERABLE_IMAGE_FORMATS.some((format) =>
                 this.name.toLowerCase().endsWith(format)
             );
-            if (!isFileRenderableImage) {
-                return undefined;
+            if (isFileRenderableImage) {
+                return this.downloadPath;
+            }
+            if (this.cloudPath.endsWith(".zarr")) {
+                try {
+                    if (this.path?.startsWith("/allen")) {
```

Comment:
I don't think the code could ever make it here for this condition because of the return on line 168 checking for the same condition.

Reply:
resolved in [552170f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/143/commits/552170f1bcfa6f438376984f8f4c54127cff0db5)[552170f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/143/commits/552170f1bcfa6f438376984f8f4c54127cff0db5)[552170f](https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/143/commits/552170f1bcfa6f438376984f8f4c54127cff0db5)